packages/eth-providers/src/base-provider.ts

```diff
@@ -523,67 +517,6 @@ export abstract class BaseProvider extends AbstractProvider {
     defineReadOnly(this, '_api', api);
   };
 
-  queryStorage = async <T = any>(
-    module: `${string}.${string}`,
-    args: any[],
-    _blockTag?: BlockTag | Promise<BlockTag> | Eip1898BlockTag
-  ): Promise<T> => {
-    const blockTag = await this._ensureSafeModeBlockTagFinalization(await parseBlockTag(_blockTag));
-    const blockHash = await this._getBlockHash(blockTag);
-
-    const registry = await this.api.getBlockRegistry(u8aToU8a(blockHash));
-
-    if (!this.storages.get(registry)) {
-      const storage = decorateStorage(
-        registry.registry,
-        registry.metadata.asLatest,
-        registry.metadata.version,
-      );
-      this.storages.set(registry, storage);
-    }
-
-    const storage = this.storages.get(registry)!;
-
-    const [section, method] = module.split('.');
-
-    const entry = storage[section][method];
-    const key = entry(...args);
-
-    const outputType = unwrapStorageType(
-      registry.registry,
-      entry.meta.type,
-      entry.meta.modifier.isOptional,
-    );
-
-    const cacheKey = `${module}-${blockHash}-${args.join(',')}`;
-    const cached = this.storageCache.get(cacheKey);
-
-    let input: Uint8Array | null = null;
-
-    if (cached) {
-      input = cached;
-    } else {
-      const value: any = await this.api.rpc.state.getStorage(key, blockHash);
-
-      const isEmpty = isNull(value);
-
-      // we convert to Uint8Array since it maps to the raw encoding, all
-      // data will be correctly encoded (incl. numbers, excl. :code)
-      input = isEmpty
-        ? null
-        : u8aToU8a(entry.meta.modifier.isOptional ? value.toU8a() : value.isSome ? value.unwrap().toU8a() : null);
-
-      this.storageCache.set(cacheKey, input);
-    }
-
-    const result = registry.registry.createTypeUnsafe(outputType, [input], {
-      blockHash,
-      isPedantic: !entry.meta.modifier.isOptional,
-    });
-
-    return result as any as T;
-  };
-
```

Comment:
https://github.com/AcalaNetwork/bodhi.js/issues/332#issuecomment-1085196293

Not sure if this will cause a timeout in getting Metadata.  Blockscout will query the account balances of many block hashes. We can run Blockscout for a few days to test it.

@ntduan please review this

Reply:
fetching metadata in parallel should already be improved https://github.com/polkadot-js/api/pull/4708, but yeah we can try and see how it actually goes